### ISSUE TITLE
Validator fails to validate a validate a valida pasted input

### ISSUE THREAD

**Issue by nssantos** [Gbuomprisco/ngx-chips#637]
PLEASE MAKE SURE THAT:
- you searched similar issues online (9/10 issues in this repo are solved by googling, so please...do it)
- you provide an online demo I can see without having to replicate your environment
- you help me by debugging your issue, and if you can't, do go on filling out this form

**I'm submitting a ...**  (check one with "x")
```
[ X ] bug report => search github for a similar issue or PR before submitting
[ ] support request/question

Notice: feature requests will be ignored, submit a PR if you'd like
```

**Current behavior**

When I try to paste a valida input (an email), the validator blocks the pasted input.
When I type the same email that I pasted, it works perfectly.

**Expected behavior**

I wanted to paste a valid email and that the validator did not fail by validating it.

**Minimal reproduction of the problem with instructions (if applicable)**

```
<tag-input
              [(ngModel)]='userEmailList'
              inputId='emailList'
              [editable]='true'
              [theme]="'t3pl-theme'"
              [inputClass]='["form-control"]'
              [addOnPaste]='true'
              secondaryPlaceholder='email@email.com'
              placeholder='+ email@email.com'
              [validators]='emailValidators'
              [modelAsStrings]='true'></tag-input>

emailValidators = [this.email]

email(control: AbstractControl): ValidationErrors|null {
    return /^(?=.{1,254}$)(?=.{1,64}@)[-!#$%&'*+/0-9=?A-Z^_`a-z{|}~]+(\.[-!#$%&'*+/0-9=?A-Z^_`a-z{|}~]+)*@[A-Za-z0-9]([A-Za-z0-9-]{0,61}[A-Za-z0-9])?(\.[A-Za-z0-9]([A-Za-z0-9-]{0,61}[A-Za-z0-9])?)*$/.test(control.value) ? null : {'email': true};
  }

```

**What do you use to build your app?. Please specify the version**

Angular-CLI 1.1.1

**Angular version:** 

Angular 4.3.4

**ngx-chips version:**

1.5.11

**Browser:** [all | Chrome XX | Firefox XX | IE XX | Safari XX | Mobile Chrome XX | Android X.X Web Browser | iOS XX Safari | iOS XX UIWebView | iOS XX WKWebView ]

**Comment by Gbuomprisco** [comment]
Hi, would you debug this line https://github.com/Gbuomprisco/ngx-chips/blob/master/modules/components/tag-input/tag-input.ts#L1051?

While debugging try to log in the console the values of `text` and `text.split(this.pasteSplitPattern)
            .map(item => this.createTag(item))`

**Comment by amozh** [comment]
@Gbuomprisco seems like the problem here is `asynchronous` execution.
Here is how `onPasteCallback` function works:
1) get pasted text
2) split text to parts with separator
3) Create and add tag for every part
3.1) create tag and do other preparation
3.2) call `onAddingRequested` -> `addItem` function
3.2.1) get display string
3.2.2) Create `Observable` and asynchronously within create `Observable`
```
Observable
            .of(model)
            .first()
            .filter(() => model.trim() !== '')
            .map(() => item)
            .map(this.createTag)
            .filter(validationFilter)
            .subscribe(subscribeFn, undefined, reset);
```
3.2.2.1) filter out empty display string (useless action because at step 3.2.2.2 model is mapped to `item`)
3.2.2.2) map observable to `item`
3.2.2.3) `createTag` again (it already happened at step 3)
3.2.2.4) validate item and filter out if invalid 
3.2.2.5) actually add item
4) emit onPaste
5) asynchronously set form input value to empty string
```
setTimeout(() => this.setInputValue(''), 0)
```

Tag is considered valid if `isTagValid` function returns `true` and `inputForm` is valid.

In my case step 5 is executed faster than step 3.2.2, hence when `validationFilter` function is called `inputForm` input is already cleared and `inputForm` is invalid, so validation fails.

Am i understanding this right way?

**Comment by na-le** [comment]
+1

**Comment by yonicozac** [comment]
+1

**Comment by rotema** [comment]
+1

**Comment by Gbuomprisco** [comment]
This should be fixed with v 1.7.1
